core/logtail.cpp

```diff
@@ -251,6 +252,7 @@ void do_worker_process() {
     if (pPlugin->LoadPluginBase()) {
         pPlugin->Resume();
     }
+    NetworkObserver::GetInstance()->Reload();
```

Comment:
It is possible to extend other observers, not just the network, so add a common observer layer here.